### PR TITLE
Add hierarchy to handlers - Widget-Control etc

### PR DESCRIPTION
**Pull request by rawagner**

### CODE REVIEW

**Comment by rawagner** [comment]
testPR

**Comment by apodhrad** [comment]
Please avoid using names such as cHandler, hwHandler, ciHandler atc...

Moreover there are lots of static references to new instances of each handler. Why did you not create a singleton for each handler?

**Comment by rawagner** [comment]
testPR

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java [hunk 1]
```diff
@@ -43,23 +44,23 @@ protected AbstractControl(T swtWidget, Runnable diagnostics) {
...
-		return controlHandler.isVisible(swtWidget);
+		return ControlHandler.getInstance().isVisible(swtWidget);
 	}
 	
 	public void setFocus(){
-		controlHandler.setFocus(swtWidget);
+		ControlHandler.getInstance().setFocus(swtWidget);
 	}
 	
 	public boolean isFocusControl(){
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: Missing JavaDoc

Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: Btw what is it? Something like hasFocus()?

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java [hunk 2]
```diff
@@ -23,18 +23,19 @@
...
 public abstract class AbstractControl<T extends org.eclipse.swt.widgets.Control> extends AbstractWidget<T> implements Control<T>{
 	
-	protected ControlHandler controlHandler = ControlHandler.getInstance();
-	
 	protected AbstractControl(Class<T> widgetClass, ReferencedComposite refComposite, int index, Matcher<?>... matchers) {
 		super(widgetClass,refComposite,index,matchers);
+		setFocus();
 	}
 
 	protected AbstractControl(T swtWidget) {
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java [hunk 3]
```diff
@@ -43,23 +44,23 @@ protected AbstractControl(T swtWidget, Runnable diagnostics) {
 	 * @return true if widget is enabled, false otherwise
 	 */
 	public boolean isEnabled(){
-		return controlHandler.isEnabled(swtWidget);
+		return ControlHandler.getInstance().isEnabled(swtWidget);
 	}
 	
 	public boolean isVisible(){
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java [hunk 4]
```diff
@@ -43,23 +44,23 @@ protected AbstractControl(T swtWidget, Runnable diagnostics) {
 	 * @return true if widget is enabled, false otherwise
 	 */
 	public boolean isEnabled(){
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java [hunk 5]
```diff
@@ -23,18 +23,19 @@
...
 		super(widgetClass,refComposite,index,matchers);
+		setFocus();
 	}
 
 	protected AbstractControl(T swtWidget) {
 		super(swtWidget);
+		setFocus();
 	}
 	
 	protected AbstractControl(T swtWidget, Runnable diagnostics) {
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java [hunk 6]
```diff
@@ -23,18 +23,19 @@
  */
 public abstract class AbstractControl<T extends org.eclipse.swt.widgets.Control> extends AbstractWidget<T> implements Control<T>{
 	
-	protected ControlHandler controlHandler = ControlHandler.getInstance();
-	
 	protected AbstractControl(Class<T> widgetClass, ReferencedComposite refComposite, int index, Matcher<?>... matchers) {
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc

File: plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/handler/ActionContributionItemHandler.java [hunk 7]
```diff
@@ -1,96 +0,0 @@
...
-import org.jboss.reddeer.core.exception.CoreLayerException;
-
-/**
- * Contains methods for handling UI operations on {@link ActionContributionItem}
- * widgets.
- * 
- * @author rawagner
- *
- */
-public class ActionContributionItemHandler {
```
Comment (apodhrad) [inline plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/handler/ActionContributionItemHandler.java]: Why did you remove this handler?

Comment (rawagner) [inline plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/handler/ActionContributionItemHandler.java]: This was used to handle view menus but I rewrite the logic for view menus in previous commits and ActionContributionItem is not used anymore anywhere

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java (hunk 1 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: this comes from eclipse swt Control API. There is not hasFocus but isFocusControl. We can discuss changing it to hasFocus in some future improvements

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java (hunk 3 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: added @override, javadoc is in Control interface

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java (hunk 4 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: added @override, javadoc is in Control interface

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java (hunk 5 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: added @override, javadoc is in Control interface

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java (hunk 2 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc added

File: plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java (hunk 6 above)
Comment (rawagner) [inline plugins/org.jboss.reddeer.swt/src/org/jboss/reddeer/swt/widgets/AbstractControl.java]: javadoc added

File: plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/handler/ActionContributionItemHandler.java (hunk 7 above)
Comment (apodhrad) [inline plugins/org.jboss.reddeer.jface/src/org/jboss/reddeer/jface/handler/ActionContributionItemHandler.java]: ok
